sonic_platform_base/component_base.py

```diff
@@ -103,3 +103,35 @@ def update_firmware(self, image_path):
             RuntimeError: update failed
         """
         raise NotImplementedError
+
+    def auto_update_firmware(self, image_path, boot_type):
+        """
+        Updates firmware of the component
+
+        This API performs firmware update automatically based on boot_type: it assumes firmware installation
+        and/or creating a loading task during the reboot, if needed, in a single call.
+        In case platform component requires some extra steps (apart from calling Low Level Utility)
+        to load the installed firmware (e.g, reboot, power cycle, etc.) - this will be done automatically during the reboot.
+        The loading task will be created by API.
+
+        Args:
+            image_path: A string, path to firmware image
+            boot_type: A string, reboot type following the upgrade
+                         - none/fast/warm/cold
+
+        Returns:
+            Output: A return code
+                return_code: An integer number, status of component firmware auto-update
+                    - return code of a positive number indicates successful auto-update
+                        - status_installed = 1
+                        - status_updated = 2
+                        - status_scheduled = 3
+                    - return_code of a negative number indicates failed auto-update
+                        - status_err_boot_type = -1
+                        - status_err_image = -2
+                        - status_err_others = -3
```

Comment:
@sujinmkang maybe it's better to use `err_unknown`?

Reply:
@nazariig  please review this PR again.